tests/unit/python/foglamp/common/test_statistics.py

```diff
@@ -0,0 +1,111 @@
+        result = {"response": "updated", "rows_affected": 1}
+        with patch.object(s._storage, 'update_tbl', return_value=result) as stat_update:
+            await s.update('READING', 5)
+            stat_update.assert_called_once_with('statistics', payload)
+            assert "updated" == result['response']
+
+    @pytest.mark.parametrize("key, value_increment, exception_name, exception_message", [
+        (123456, 120, TypeError, "key must be a string"),
+        ('PURGED', '120', ValueError, "value must be an integer"),
+        (None, '120', TypeError, "key must be a string"),
+        ('123456', '120', ValueError, "value must be an integer"),
+        ('READINGS', None, ValueError, "value must be an integer")
+    ])
+    async def test_update_with_invalid_params(self, key, value_increment, exception_name, exception_message):
+        storage_client_mock = MagicMock(spec=StorageClient)
+        s = Statistics(storage_client_mock)
+
+        with pytest.raises(exception_name) as excinfo:
+            await s.update(key, value_increment)
+
+        assert exception_message == str(excinfo.value)
+
+    async def test_update_exception(self):
+        storage_client_mock = MagicMock(spec=StorageClient)
+        s = Statistics(storage_client_mock)
+        msg = 'Unable to update statistics value based on statistics_key %s and value_increment %d,' \
+              ' error %s', 'BUFFERED', 5, ''
+        with patch.object(s._storage, 'update_tbl', side_effect=Exception()):
+            with pytest.raises(Exception):
+                with patch.object(_logger, 'exception') as logger_exception:
+                    await s.update('BUFFERED', 5)
+            logger_exception.assert_called_once_with(*msg)
+
+    async def test_add_update(self):
+        stat_dict = {'FOGBENCH/TEMPERATURE': 1}
+        storage_client_mock = MagicMock(spec=StorageClient)
+        s = Statistics(storage_client_mock)
+        payload = '{"where": {"column": "key", "condition": "=", "value": "FOGBENCH/TEMPERATURE"}, ' \
+                  '"expressions": [{"column": "value", "operator": "+", "value": 1}]}'
+        result = {"response": "updated", "rows_affected": 1}
+        with patch.object(s._storage, 'update_tbl', return_value=result) as stat_update:
+            await s.add_update(stat_dict)
+            stat_update.assert_called_once_with('statistics', payload)
+            assert "updated" == result['response']
+
+    async def test_insert_when_key_error(self):
+        stat_dict = {'FOGBENCH/TEMPERATURE': 1}
+        storage_client_mock = MagicMock(spec=StorageClient)
+        s = Statistics(storage_client_mock)
+        payload = '{"previous_value": 0, "value": 1, "key": "FOGBENCH/TEMPERATURE", ' \
+                  '"description": "The number of readings received by FogLAMP since startup' \
+                  ' for sensor FOGBENCH/TEMPERATURE"}'
+        result = {"response": "inserted", "rows_affected": 1}
```

Comment:
better to rename it as expected or expected_result. in general  its for actual result